scripts/docker-run.sh

```diff
@@ -38,6 +38,27 @@ else
 	DOCKER_RUN="docker run"
 fi
 
+#
+# Ensure that either UPSTREAM_PRODUCT_BRANCH or DELPHIX_PACKAGE_MIRROR_MAIN
+# is set. If neither is set, attempt to use git to determine the upstream
+# branch. If that also fails, then exit.
+#
+UPSTREAM_BRANCH="$UPSTREAM_PRODUCT_BRANCH"
```

Comment:
Note that we are also using `UPSTREAM_PRODUCT_BRANCH` in `scripts/build-ancillary-repository.sh`. I think it would be better to set a default variable for the same variable if it is empty instead of creating a new variable.